tests/test_capability.py

```diff
@@ -0,0 +1,287 @@
+from utility import repoutil
+
+
+class TestCapabilityClass(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(TestCapabilityClass, self).__init__(*args, **kwargs)
+
+        # Declare variables
+        self.tmp_dir = '{}/'.format(yc_gc.temp_dir)
+        self.yangcatalog_api_prefix = '{}/api/'.format(yc_gc.my_uri)
+        self.prepare_output_filename = 'prepare'
+        self.hello_message_filename = 'capabilities-ncs5k.xml'
+        self.platform_name = 'ncs5k'
+        self.resources_path = '{}/resources'.format(os.path.dirname(os.path.abspath(__file__)))
+
+    #########################
+    ### TESTS DEFINITIONS ###
+    #########################
+
+    def test_capability_parse_and_dump_sdo(self):
+        """
+        Test if keys were created and prepare object values were set correctly from all .yang files which are located in 'path' directory.
+        """
+        repo = self.get_yangmodels_repository()
+        path = '{}/standard/ieee/published/802.3'.format(yc_gc.yang_models)
+        api = False
+        sdo = True
+        prepare = Prepare(yc_gc.logs_dir, self.prepare_output_filename, self.yangcatalog_api_prefix)
+
+        capability = Capability(yc_gc.logs_dir, path, prepare,
+                                None, api, sdo, self.tmp_dir, yc_gc.result_dir,
+                                yc_gc.save_file_dir, yc_gc.private_dir, yc_gc.yang_models)
+
+        capability.parse_and_dump_sdo(repo)
+
+        for root, subdirs, sdos in os.walk(path):
+            for file_name in sdos:
+                if '.yang' in file_name and ('vendor' not in root or 'odp' not in root):
+                    path_to_yang = '{}/{}'.format(path, file_name)
+                    yang = self.declare_sdo_module(path_to_yang)
+                    key = '{}@{}/{}'.format(yang.name, yang.revision, yang.organization)
+                    self.assertIn(key, capability.prepare.yang_modules)
+
+    def test_capability_parse_and_dump_sdo_api(self):
+        """
+        Test if key was created and prepare object value was set correctly from all modules loaded from prepare-sdo.json file.
+        """
+        repo = self.get_yangmodels_repository()
+
+        path = '{}/tmp/capability-tests/temp'.format(self.resources_path)
+        json_dir = '{}/tmp/capability-tests'.format(self.resources_path)
+        api = True
+        sdo = True
+
+        prepare = Prepare(yc_gc.logs_dir, self.prepare_output_filename, self.yangcatalog_api_prefix)
+
+        capability = Capability(yc_gc.logs_dir, path, prepare,
+                                None, api, sdo, json_dir, yc_gc.result_dir,
+                                yc_gc.save_file_dir, yc_gc.private_dir, yc_gc.yang_models)
+
+        capability.parse_and_dump_sdo(repo)
+
+        with open('{}/prepare-sdo.json'.format(json_dir), 'r') as f:
+            sdos_json = json.load(f)
+
+        sdos_list = sdos_json['modules']['module']
+        for sdo in sdos_list:
```

Comment:
test if the length is non zero as well so we don t end up with valid test and nothing in module list.. same in all cases